### ISSUE TITLE
Swift 3 support?

### ISSUE THREAD

**Issue by swabzz** [FuckBoilerplate/base_app_ios#1]
Are you going to update boilerplate to swift 3??

**Comment by robertofrontado** [comment]
@bagwanshoaib Yes, of course! 

Currently I'm migrating some of my libraries to Swift 3, and I will migrate this one too but after I finish the others

But if you want to give it a try, feel free 👍

**Comment by swabzz** [comment]
Any update on this?

**Comment by robertofrontado** [comment]
Sadly not yet, still working on the others libraries migration, hope to get this working soon

**Comment by swabzz** [comment]
@robertofrontado do you want me to do it for you?

Let me know the way

**Comment by robertofrontado** [comment]
Hello @swabzz,

I already migrated it but it is missing the [RxGcm](https://github.com/robertofrontado/RxGcm-iOS) integration (Since I didn't migrate it yet) and the `freshStart` functionality, anyway you can find the migration in this [branch](https://github.com/FuckBoilerplate/base_app_ios/tree/swift_3), I hope I can get it working very soon.

Sorry if this has taken too long.

**Comment by swabzz** [comment]
Hello robert

Why we are using RXGCM, instead we can use APNS ?

**Comment by robertofrontado** [comment]
@swabzz Yeah sure! You don't have to use  [RxGcm](https://github.com/robertofrontado/RxGcm-iOS) , you can use APNS without problems,

But we are using [RxGcm](https://github.com/robertofrontado/RxGcm-iOS) here because we were using [Google Cloud Messaging](https://developers.google.com/cloud-messaging/ios/client) (I'm aware I have to develop a new library call RxFcm, in order to use [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/ios/client)), and to use Gcm you must do some configuration in your AppDelegate, so I just wanted to avoid to add that boilerplate in each application I make, 

Also [RxGcm](https://github.com/robertofrontado/RxGcm-iOS) provides you extra functionalities,

```
Features:
- Remove iOS boilerplate code (not need to do the dirty configuration in the AppDelegate).
- Decouple presentation responsibilities from data responsibilities when receiving notifications.
- Deploy a targeting strategy to aim for the desired ViewController when receiving notifications.
```

So basically you can just conform `GcmReceiverUIForeground` protocol in one of your `ViewControllers` and it will receives the notification if the result of `func matchesTarget(key: String) -> Bool` is true, allowing you to refresh the data showed in that `ViewController`,

Hope this helps!

**Comment by swabzz** [comment]
Yes @robertofrontado that would be awesome if we go for FCM And thanks for explaining things

**Comment by robertofrontado** [comment]
@swabzz Thanks, I'm going to work on it hopefully this week, and don't worry! I'm glad to help 👍

**Comment by swabzz** [comment]
Thanks for expaining dear

Get Outlook for iOShttps://aka.ms/o0ukef

On Wed, Oct 26, 2016 at 6:10 AM +0530, "Roberto Frontado" <notifications@github.com<mailto:notifications@github.com>> wrote:

@swabzzhttps://github.com/swabzz Yeah sure! You don't have to use RxGcmhttps://github.com/robertofrontado/RxGcm-iOS , you can use APNS without problems,

But we are using RxGcmhttps://github.com/robertofrontado/RxGcm-iOS here because we were using Google Cloud Messaginghttps://developers.google.com/cloud-messaging/ios/client (I'm aware I have to develop a new library call RxFcm, in order to use Firebase Cloud Messaginghttps://firebase.google.com/docs/cloud-messaging/ios/client, and to use Gcm you must do some configuration in your AppDelegate, so I just wanted to about to add that boilerplate in each application I made,

RxGcmhttps://github.com/robertofrontado/RxGcm-iOS provides you extra functionalities,

Features:
- Remove iOS boilerplate code (not need to do the dirty configuration in the AppDelegate).
- Decouple presentation responsibilities from data responsibilities when receiving notifications.
- Deploy a targeting strategy to aim for the desired ViewController when receiving notifications.

So basically you can just conform GcmReceiverUIForeground protocol in one of your ViewControllers and it will receives the notification if the result of func matchesTarget(key: String) -> Bool is true, allowing you to refresh the data showed in that ViewController,

Hope this helps!

##
